diff --git a/src/SDL_ttf.c b/src/SDL_ttf.c
--- a/src/SDL_ttf.c
+++ b/src/SDL_ttf.c
@@ -1188,7 +1188,10 @@ static bool Render_Line_##NAME(TTF_Font *font, SDL_Surface *textbuf, int xstart,
             int remainder;                                                                                              \
             Uint8 *saved_buffer = image->buffer;                                                                        \
             int saved_width = image->width;                                                                             \
+            if (image->buffer == NULL) {                                                                                \
+                continue;                                                                                               \
+            }                                                                                                           \
             image->buffer += alignment;                                                                                 \
             /* Position updated after glyph rendering */                                                                \
             x = xstart + FT_FLOOR(x) + image->left;                                                                     \
             y = ystart + FT_FLOOR(y) - image->top;                                                                      \
